lib/react_test_utils.dart

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) 2013, the Clean project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library react_test_utils;
+
+import "package:react/react_client.dart" show ReactComponentFactory, ReactComponentFactoryProxy;
+import "dart:js";
+import "dart:html" show Element;
+
+_getNestedJsObject(JsObject base, List<String> keys, [String errorIfNotFound='']) {
+  JsObject object = base;
+  for (String key in keys) {
+    if (!object.hasProperty(key)) {
+      throw 'Unable to resolve $key in $base${keys.join('.')}}.\n$errorIfNotFound';
+    }
+    object = object[key];
+  }
+  return object;
+}
+
+JsObject _ReactTestUtils = _getNestedJsObject(context, ['React', 'addons', 'TestUtils'],
+  'React.addons.TestUtils not found. Ensure you\'ve included the React addons in your HTML file.'
+  '\n  This:\n<script src="packages/react/react-with-addons.js"></script>'
+  '\n  Not this:\n<script src="packages/react/react.js"></script>');
+
+JsFunction getFactory(ReactComponentFactory reactComponentFactory) {
+  if (reactComponentFactory is ReactComponentFactoryProxy) {
+    return (reactComponentFactory as ReactComponentFactoryProxy).reactComponentFactory;
+  }
+  return null;
+}
+
+typedef bool ComponentTestFunction(JsObject componentInstance);
+
+class ReactTestUtils {
```

Comment:
This class could be done away with, in favor of top-level functions (still thinking of packages in the JS way).